frontend/src/hooks/useCriteriaChartData.ts

```diff
@@ -0,0 +1,55 @@
+import { useMemo } from 'react';
+import { VideoSerializerWithCriteria } from 'src/services/openapi';
+import usePersonalCriteriaScores from 'src/hooks/usePersonalCriteriaScores';
+
+const between = (a: number, b: number, x: number | undefined): number => {
+  // clips x between a and b
+  return Math.min(b, Math.max(a, x || 0));
+};
+
+const SCORE_MIN = -1;
+const SCORE_MAX = 1;
+const DOMAIN = [SCORE_MIN, SCORE_MAX];
+
+const useCrtiteriaChartData = (video: VideoSerializerWithCriteria) => {
+  const { criteria_scores: criteriaScores } = video;
+  const shouldDisplayChart = criteriaScores && criteriaScores.length > 0;
+
+  const { personalScoresActivated, personalCriteriaScores } =
+    usePersonalCriteriaScores();
+
+  const personalScoreByCriteria = useMemo(() => {
+    if (!personalScoresActivated || personalCriteriaScores === undefined)
+      return {};
+
+    const result: { [criteria: string]: number | undefined } = {};
+    personalCriteriaScores.forEach(
+      ({ criteria, score }) => (result[criteria] = score)
+    );
+    return result;
+  }, [personalScoresActivated, personalCriteriaScores]);
+
+  const data = useMemo(() => {
+    if (!shouldDisplayChart) return [];
+
+    return criteriaScores
+      .filter((s) => s.criteria != 'largely_recommended')
+      .map(({ score, criteria }) => {
+        const clippedScore = between(SCORE_MIN, SCORE_MAX, score);
+        return {
+          criteria,
+          clippedScore,
+          personalScore: personalScoreByCriteria[criteria],
+        };
```

Comment:
We either need to:
1. Fix the scale (from -1 to 1) on the BarChart and Radar charts. In this case we could even not have clippedScores.
2. Have `clippedPersonalScore` added in the data for the display

Otherwise when have a personnal score > 1.0 (displayed as >10.0) we obtain for example:

![Screenshot 2022-04-22 at 10 30 22](https://user-images.githubusercontent.com/8818081/164651860-bba9f5af-b657-4998-8535-a6a389fd525e.png)


It is important that the scale is fixed such that all the BarCharts and RadarChart from one video to another are easily comparable. It is not a worry that we lose the information when the score if >1

Reply:
I added a `clippedPersonalScore` and used it as value.

How can I get a video with a personal score > 1.0?

Reply:
> How can I get a video with a personal score > 1.0?

If you have access to the Django admin, you can manually change the ContributorRatingCriteriaScore in it.